js/src/Ice/TcpEndpointI.js

```diff
@@ -7,33 +7,36 @@ import { StringUtil } from "./StringUtil.js";
 import { TCPEndpointInfo } from "./Endpoint.js";
 import { ParseException } from "./LocalExceptions.js";
 import { IPEndpointI } from "./IPEndpointI.js";
-import { TcpTransceiver } from "./TcpTransceiver.js";
 import { EndpointInfo as SSLEndpointInfo } from "./SSL/EndpointInfo.js";
+import { TcpTransceiver } from "./TcpTransceiver.js";
 
 export class TcpEndpointI extends IPEndpointI {
-    constructor(instance, ho, po, sif, ti, conId, co) {
-        super(instance, ho, po, sif, conId);
+    constructor(instance, host, port, sourceAddress, timeout, connectionId, compress) {
+        super(instance, host, port, sourceAddress, connectionId);
         // The default timeout is 60,000 milliseconds (1 minute). It's not used in Ice 3.8 or greater.
-        this._timeout = ti === undefined ? (instance ? 60000 : undefined) : ti;
-        this._compress = co === undefined ? false : co;
+        this._timeout = timeout === undefined ? (instance ? 60000 : undefined) : timeout;
+        this._compress = compress === undefined ? false : compress;
     }
 
     //
     // Return the endpoint information.
     //
     getInfo() {
-        const info = new TCPEndpointInfo();
-        this.fillEndpointInfo(info);
+        let info = new TCPEndpointInfo(
+            this._timeout,
+            this._compress,
+            this._host,
+            this._port,
+            this._sourceAddr,
+            this.type(),
+            this.secure(),
+        );
 
         if (this.secure()) {
-            const sslInfo = new SSLEndpointInfo();
-            sslInfo.underlying = info;
-            sslInfo.timeout = info.timeout;
-            sslInfo.compress = info.compress;
-            return sslInfo;
-        } else {
-            return info;
+            info = new SSLEndpointInfo(info);
```

Comment:
It is tested with the browser WSS protocol. In this case is used as the underlying endpoint info for WSSEndpointInfo.

The JavaScript SSLEndpointInfo doesn't have additional fields because they are not accessible with the WebSocket API.

In JavaScript for SSLEndpointInfo, and SSLConnectionInfo we just use the constructor inherited from Ice.EndpointInfo and create them with an underlying.

Reply:
We should look into adding browsers testing with CI